test/e2e/sagemaker/tests/test_transformjob.py

```diff
@@ -0,0 +1,154 @@
+
+
+@service_marker
+@pytest.mark.canary
+class TestTransformJob:
+    def _get_created_transformjob_status_list(self):
+        return ["InProgress", "Completed"]
+    
+    def _get_stopped_transformjob_status_list(self):
+        return ["Stopped", "Stopping"]
+
+    def _get_resource_transformjob_arn(self, resource: Dict):
+        assert 'ackResourceMetadata' in resource['status'] and \
+            'arn' in resource['status']['ackResourceMetadata']
+        return resource['status']['ackResourceMetadata']['arn']
+
+    def _get_sagemaker_transformjob_arn(self, sagemaker_client, transformjob_name: str):
+        try:
+            transformjob = sagemaker_client.describe_transform_job(TransformJobName=transformjob_name)
+            return transformjob['TransformJobArn']
+        except BaseException:
+            logging.error(
+                f"SageMaker could not find a transformJob with the name {transformjob_name}")
+            return None
+
+    def _get_sagemaker_transformjob_status(self, sagemaker_client, transformjob_name: str):
+        try:
+            transformjob = sagemaker_client.describe_transform_job(TransformJobName=transformjob_name)
+            return transformjob['TransformJobStatus']
+        except BaseException:
+            logging.error(
+                f"SageMaker could not find a transformJob with the name {transformjob_name}")
+            return None
+
+    def test_create_transformjob(self, xgboost_transformjob):
+        (reference, resource) = xgboost_transformjob
+        assert k8s.get_resource_exists(reference)
+
+    def test_transformjob_has_correct_arn(self, sagemaker_client, xgboost_transformjob):
+        (reference, _) = xgboost_transformjob
+        resource = k8s.get_resource(reference)
+        transformjob_name = resource['spec'].get('transformJobName', None)
+
+        assert transformjob_name is not None
+
+        resource_transformjob_arn = self._get_resource_transformjob_arn(resource)
+        expected_transformjob_arn = self._get_sagemaker_transformjob_arn(
+            sagemaker_client, transformjob_name)
+
+        assert resource_transformjob_arn == expected_transformjob_arn
+
+    def test_transformjob_has_created_status(self, sagemaker_client, xgboost_transformjob):
+        (reference, _) = xgboost_transformjob
+        resource = k8s.get_resource(reference)
+        transformjob_name = resource['spec'].get('transformJobName', None)
+
+        assert transformjob_name is not None
+
+        current_transformjob_status = self._get_sagemaker_transformjob_status(
+            sagemaker_client, transformjob_name)
+        expected_transformjob_status_list = self._get_created_transformjob_status_list()
+        assert current_transformjob_status in expected_transformjob_status_list
+        
+
+    def test_transformjob_has_stopped_status(self, sagemaker_client, xgboost_transformjob):
+        (reference, _) = xgboost_transformjob
+        resource = k8s.get_resource(reference)
```

Comment:
@RedbackThomson do you have any suggestion to avoid this duplicate code in every test. I also faced this issue

dont need to change this PR, just looking for options

Reply:
Um well it is kind of necessary to get the latest state of the resource. If we did a `get_resource` within the fixture, since the fixture is only loaded once (`scope="module"`), we would not see it updating over time.

That being said, the fixture is already returning a resource with the spec. I think most of the time when we are calling `get_resource`, it's not necessary, since we are just looking at items in the spec fields. We only need to call `get_resource` if we want to see the `status`.

Reply:
Sg